### PR TITLE
Add ability to run on selected filters

### PR DESCRIPTION
**Pull request by ghost**
FILTERS directive in settings.js now allows user to specify what filters to use. When empty or not specified, vegassist will pick up all filters in directory.

### CODE REVIEW

**Comment by squeek502** [comment]
Looks good to me. Did you forget to include vegassist.js changes in your commit? With only the changes in this PR, the FILTERS setting isn't actually being used.

File: lib/filter.js [hunk 1]
```diff
@@ -12,13 +12,35 @@ var TweetFilter = function(filters_or_dirpath, excluded_terms) {
...
-TweetFilter.getFiltersFromDirectory = function(dir) {
+TweetFilter.getFiltersFromDirectory = function(dir, filters) {
     dir = path.resolve(dir);
     var filter_files = fs.readdirSync(dir);
     var joined_files = [];
-    filter_files.forEach(function(file) {
-        joined_files.push(path.join(dir, file));
-    })
+
+    if( filters.length == 0 ) {
```
Comment (squeek502) [inline lib/filter.js]: This should be `if ( !filters || filters.length == 0 ) {` to allow for a null filters being passed in.

**Comment by squeek502** [comment]
My recommended `vegassist.js` changes:

``` diff
 // Load filters from all files in the filters directory
-var filter = new TweetFilter('filters', settings.FILTERED_TERMS);
+var filter = new TweetFilter('filters', settings.FILTERED_TERMS, settings.FILTERS);
```

``` diff
 fs.appendFile(logFile, "", function(err) {
     if (err) { console.warn("Unable to write to log file: " + path.relative(__dirname, logFile)); }
 })

+console.log("Loaded filters: " + Object.keys(filter.filters).join(", "))
+
 // log tweets and their matches as json, each tweet/match will be a separate line of json
 var logMatches = function(tweet, matches) {
```

**Comment by ghost** [comment]
You are right, I forgot to commit vegassist.js. I ensured settings.FILTERS is being passed in (always as an array). The function getFiltersFromDirectory now also handles null parameters.

**Comment by squeek502** [comment]
Nice work! :+1:

**Comment by ghost** [comment]
Thank you mate! :+1:

**Pull request merged**